### PR TITLE
Thai/review spec tests

### PR DESCRIPTION
**Pull request by athai**
fixes issue #99

## Changes in this PR.

- [X] Review tests for correctness
- [ ] Rewrite tests to actually check for things

## Testing this PR.

Currently N/A.

### Expected Output.

```
$ rake spec
[...]
```

### CODE REVIEW

**Comment by athai** [comment]
In the spec tests for DiskSize, DiskTemplate, and VCPUCount, ``(...).to not_exist`` is being used instead of ``(...).to_not exist``. Everywhere else, ``(...).to/to_not exist`` is being used instead. What's the difference?

**Comment by alxngyn** [comment]
So I found out that some of these are passing because they're not being ran at all. I'm fixing them up and will push to this branch when I get through all of them.

Good work @athai !

File: spec/models/client_spec.rb [hunk 1]
```diff
@@ -49,7 +49,9 @@ def app
     client = Client.create(name: 'Delete Me')
     expect(client).to exist
     expect { client.delete }.to_not raise_error
+    ### Test passes but the item's name and address can still be printed?
     expect(client).to_not exist
+    print client, ' ', client.name
```
Comment (athai) [inline spec/models/client_spec.rb]: Is this normal behavior?

Comment (alxngyn) [inline spec/models/client_spec.rb]: I'm 80% sure so. I think the behind the scenes logic is like this:

1. var client is stored in memory to a Client object with name = 'Delete Me'
2. client.delete is ran which is just a SQL execution on the DB
3. the print line is referring data from the variable in the memory and not the DB record.

I think the problem is the `().to exists` and `().to_not exist` which is doing something else than we are expecting.

Comment (Kennric) [inline spec/models/client_spec.rb]: Yes, I think that is correct, Model.delete removes the database entry and returns self (the object you are trying to delete). Probably the way to test this is to try and find the object in the database again and test that it isn't there anymore - or, use Sequel's model method 'exists?' which tests if the object exists in the database (expect client.exists? to eq false)

Comment (athai) [inline spec/models/client_spec.rb]: Is ``expect(client).to_not exist`` logically equivalent to ``expect(client.exists?).to be false``?

Comment (alxngyn) [inline spec/models/client_spec.rb]: From what I could google, they are not equivalent:

> `expect(client).to_not exist` checks if client is falsey (nil or false)
> ```
> From rspec docs:
> expect(actual).to exist        # passes if actual.exist? and/or actual.exists? are truthy
> ```
> https://www.relishapp.com/rspec/rspec-expectations/docs/built-in-matchers

> `expect(client.exists?).to be false` checks if the result from `client.exists?` is false (not falsey)
> ```
> From rspec docs:
> expect(actual).to be false     # passes if actual == false
> ```
> https://www.relishapp.com/rspec/rspec-expectations/docs/built-in-matchers

> 
> `client.exists?` should return true or false depending if client exists
> ```
> From rspec docs: 
> The exist matcher is used to specify that something exists
> (as indicated by #exist? or #exists?):
> ```
> https://www.relishapp.com/rspec/rspec-expectations/v/2-5/docs/built-in-matchers/exist-matcher

**Comment by Kennric** [comment]
That looks pretty good, tests are running, and fail when I modify them slightly, as they should. +1

**Comment by alxngyn** [comment]
The tests still pass but the fixes work as they should. Good work. +1

**Pull request merged**